src/app/qgsattributetabledialog.cpp

```diff
@@ -351,6 +353,26 @@ void QgsAttributeTableDialog::updateFieldFromExpressionSelected()
   runFieldCalculation( mLayer, mFieldCombo->currentText(), mUpdateExpressionText->currentField(), filteredIds );
 }
 
+void QgsAttributeTableDialog::viewWillShowContextMenu( QMenu* menu, QModelIndex atIndex )
+{
+  if ( menu )
+  {
+    menu->addAction( tr( "Zoom to feature" ), this, SLOT( zoomToFeature() ) );
```

Comment:
> Would here be the place to add a check for geometryless layers

Good point, but a better check would be whether the current feature itself has a non-null geometry. That would also handle spatial layers where an individual feature doesn't have geometry.